tutorials/tutorial2.md

```diff
@@ -0,0 +1,152 @@
+
+![subsampled_NK_cells](NK.2400.50.png)
+
+Compared to the 'clean' signal from 2400 cells, the aggregated ATAC-seq profile of these 50 cells is noisy. Because the signal is noisy, peak calls calculated by MACS2 on this data (shown as red bars below the signal tracks) are also inaccurate. The AUPRC of peak calling by MACS2 on the noisy data is only 0.20.
+
+As reported in our paper, we trained an AtacWorks model to learn a mapping from 50-cell signal to 2400-cell signals and peak calls. In other words, given a noisy ATAC-seq signal from 50 cells, this model learned what the signal would look like - and where the peaks would be called - if we had sequenced 2400 cells. This model was trained on data from Monocytes and B cells, so it has not encountered data from NK cells.
+
+If you want to train your own AtacWorks model instead of using the model reported in the paper, refer to [Tutorial 1](tutorial1.md).
+
+
+## Step 1: Set parameters
+
+```
+atacworks=<path to atacworks>
+```
+
+## Step 2: Download model
+
+Download a pre-trained deep learning model (model.pth.tar) trained with dsc-ATAC-seq data from Monocytes and B cells. This model was reported and used in the AtacWorks paper (1).
+```
+mkdir models
+wget -P models https://atacworks-paper.s3.us-east-2.amazonaws.com/dsc_atac_blood_cell_denoising_experiments/50_cells/models/model.pth.tar
+```
+
+## Step 3: Download config files
+
+We also need to download the 'configs' directory containing config files for this experiment. The config files describe the structure of the deep learning model and the parameters used to run inference.
+```
+mkdir configs
+wget -P configs https://atacworks-paper.s3.us-east-2.amazonaws.com/dsc_atac_blood_cell_denoising_experiments/50_cells/configs/infer_config.yaml
+wget -P configs https://atacworks-paper.s3.us-east-2.amazonaws.com/dsc_atac_blood_cell_denoising_experiments/50_cells/configs/model_structure.yaml
+```
+
+## Step 4: Download the test dsc-ATAC-seq signal from 50 NK cells (~1M reads), in bigWig format
+
+```
+wget https://atacworks-paper.s3.us-east-2.amazonaws.com/dsc_atac_blood_cell_denoising_experiments/50_cells/test_data/noisy_data/dsc.1.NK.50.cutsites.smoothed.200.bw
+```
+
+## Step 5: Create genomic intervals to define regions for testing
+
+The model we downloaded takes the input ATAC-seq signal in non-overlapping genomic intervals spanning 50,000 bp. To define the genomic regions for the model to read, we take the chromosomes on which we want to apply the model and split their lengths into 50,000-bp intervals, which we save in BED format. 
+In this example, we will apply the model to chromosomes 1-22. The reference genome we use is hg19. We use the prepared chromosome sizes file `hg19.auto.sizes`, which contains the sizes of chromosomes 1-22 in hg19.
+```
+python $atacworks/scripts/get_intervals.py \
+    --sizes $atacworks/example/reference/hg19.auto.sizes \
+    --intervalsize 50000 \
+    --out_dir intervals \
```

Comment:
We need to do `mkdir intervals` before this step. Otherwise, I'm getting this error : 
```
(atacworks-3.7) ntadimeti@atacworks-ntadimeti:~/AtacWorks$ python $atacworks/scripts/get_intervals.py     --sizes $atacworks/example/reference/hg19.auto.sizes     --intervalsize 50000     --out_dir /home/ntadimeti/AtacWorks/intervals     --prefix hg19.50000     --wg
INFO:2020-03-05 16:30:52,017:AtacWorks-intervals] Generating intervals tiling across all chromosomes             in sizes file: /home/ntadimeti/AtacWorks/example/reference/hg19.auto.sizes
Traceback (most recent call last):
  File "/home/ntadimeti/AtacWorks/scripts/get_intervals.py", line 226, in <module>
    main()
  File "/home/ntadimeti/AtacWorks/scripts/get_intervals.py", line 158, in main
    df_to_bed(intervals, out_file_path)
  File "/home/ntadimeti/miniconda3/envs/atacworks-3.7/lib/python3.7/site-packages/claragenomics/io/bedio.py", line 69, in df_to_bed
    df.to_csv(out_file_path, sep='\t', index=False, header=header)
  File "/home/ntadimeti/miniconda3/envs/atacworks-3.7/lib/python3.7/site-packages/pandas/core/generic.py", line 3229, in to_csv
    formatter.save()
  File "/home/ntadimeti/miniconda3/envs/atacworks-3.7/lib/python3.7/site-packages/pandas/io/formats/csvs.py", line 183, in save
    compression=self.compression,
  File "/home/ntadimeti/miniconda3/envs/atacworks-3.7/lib/python3.7/site-packages/pandas/io/common.py", line 397, in _get_handle
    f = open(path_or_buf, mode, encoding=encoding, newline="")
FileNotFoundError: [Errno 2] No such file or directory: '/home/ntadimeti/AtacWorks/intervals/hg19.50000.genome_intervals.bed'
```

Reply:
Added